### PR TITLE
various changes

### PR DESCRIPTION
**Pull request by DavidBJaffe**
D GENES AND JUNCTION REGIONS
- detect VDDJ
- add ALIGNn, that exhibits a visual alignment for chain n (for each exact subclonotype) to the 
  donor V(D)J reference, picking the best D for heavy chains / TRB; there are also ALIGN_2NDn, 
  JALIGNn, and JALIGN_2NDn.
- add GROUP_VJ_REFNAME_HEAVY and GROUP_VDJ_REFNAME_HEAVY
- document opt_d,... and change name to. d1_name,...
- add d1_score and d2_score
- improve opt_d
- add notion of global variables
- add global variables d_inconsistent_% and d_inconsistent_n
- add filter D_INCONSISTENT to show only clonotypes having D gene assignment inconsistencies
- add filters D_NONE and D_SECOND
- add new page on site "D genes and junction regions"

CRATES
- update crates
- move initial processing into its own crate (weirdly this made the executable 2% larger)
- relax crate dependency restriction, so that enclone_core can depend on enclone_proto
- import a subset of rust-bio, with one modification

COMPUTATIONAL PERFORMANCE
- split some timers
- speed up doublet filtering and VDJ/GEX inconsistency testing

TESTING
- put all the tests that are affected by D region algorithms in their own test category
- ./test now fails if it used too much time
- move unaccounted time test prevent concurrence and remove from GitHub Actions
- don't warn about unused imports in tests

OTHER
- add a script that tags an enclone version corresponding to a given cellranger version
- add split-debuginfo = "unpacked" to Cargo.toml, because it speeds up Mac compilation
- add undocumented option TOOLTIP to add tooltip text to honeycomb plots
- landing pages now exhibits all help pages in a two-column (more compact) format
- add links to previous releases matching particular Cell Ranger versions
- fix bug in SUBSET_JSON
- add filter MAX_EXACTS

### CODE REVIEW

**Review by wyattmcdonnell — APPROVED** [review]
Closing my review on this after latest changes to command line exposure of D gene options at 2:01 pm. Nicely done, thanks for your patience! No issues with any tests on my side as of `08a9f58`. Hopefully a while before this many sweeping changes land at once again 😅

File: pages/index.html.src [hunk 1]
```diff
@@ -531,25 +543,34 @@ For completeness, here are all the other pages on this site:
...
+    </a></td> <td align="left">those curious what install command does</td>
 </tr>
 <tr>
     <td align="left"><a href="pages/auto/innate.html">iNKT and MAIT cells
     </a></td> <td align="left">people interested in iNKT and MAIT cells</td>
 </tr>
 <tr>
     <td align="left"><a href="pages/auto/features.html">V(D)J features
-    </a></td> <td align="left">people interested in how CDR1, CDR2 etc. are calculated</td>
+    </a></td> <td align="left">those curious about CDR*, etc. calculations</td>
```
[the remark this answers is not in the document]
Comment (DavidBJaffe) [reply pages/index.html.src:554]: I had to cut characters to get everything to fit side by side on one page.

[the remark this answers is not in the document]
Comment (DavidBJaffe) [reply pages/index.html.src:554]: Making the first change.

**Comment by DavidBJaffe** [comment]
Thank you for reviewing this monstrosity!

**Pull request merged**